### PR TITLE
Remove bindings

### PR DESCRIPTION
**Pull request by pnmadelaine**
## Proposed changes

This PR removes the OCaml and JS bindings since they were moved to `cryspen/hacl-packages`.

## Types of changes

- [ ] Proof maintenance (non-breaking change which fixes a proof regression)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New algorithm or feature (non-breaking change which adds functionality)
- [ ] Improved performance (fix or feature that would improve performance of some algorithm on some platform)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [x] I have read the [CODE_OF_CONDUCT.md](https://github.com/hacl-star/hacl-star/blob/main/CODE_OF_CONDUCT.md) doc
- [x] I have read and agree to submit my changes under the [LICENSE](https://github.com/hacl-star/hacl-star/blob/main/LICENSE)
- [x] I have added necessary documentation (if appropriate)
- [x] I have edited [CHANGES.md](https://github.com/hacl-star/hacl-star/blob/main/CHANGES.md) (if appropriate)

### CODE REVIEW

File: doc/Obtaining.rst [hunk 1]
```diff
@@ -88,43 +88,12 @@ and other HACL* users have also developed bindings for other programming languag
...
-``opam install hacl-star-raw``.
-
-From source, the low-level bindings can be installed by running
-``make install-hacl-star-raw`` in ``dist/gcc-compatible``. Then, the high-level
-wrapper can be installed by running ``dune build && dune install`` in
-``bindings/ocaml``.
-
-Documentation for the high-level API is available `here
-<https://hacl-star.github.io/ocaml_doc/>`_.
+OCaml bindings were moved to the `cryspen/hacl-packages` repository.
```
Comment (franziskuskiefer) [inline doc/Obtaining.rst:91]: There's https://tech.cryspen.com/hacl-packages/hacl-js/index.html and https://tech.cryspen.com/hacl-packages/hacl-ocaml/index.html.
Can you add documentation that you think makes sense there?

File: dist/gcc-compatible/configure [hunk 2]
```diff
@@ -227,7 +227,6 @@ show_help () {
   printf "                    * aarch64-apple-ios\n"
   printf "                    * x86_64-apple-ios-simulator\n"
   printf "    --disable-bzero Do not use explicit_bzero (binary will work with an old GLIBC)\n"
-  printf "    --disable-ocaml Disable OCAML bindings\n"
```
Comment (franziskuskiefer) [inline dist/gcc-compatible/configure:230]: I'm not sure whether some folks would prefer keeping the option to build the raw package here.
If yes, the `hacl-star-raw.opam` should be thinned out to make clear that it's not the real one and we should keep the infrastructure here.
But I'd be fine with removing this here as you've done.

**Review by franziskuskiefer — APPROVED** [review]
Thanks, this generally looks good to me with an open question and a follow up.

File: doc/Obtaining.rst (hunk 1 above)
Comment (pnmadelaine) [reply doc/Obtaining.rst:91]: I think the links you provided should be the entry points to the documentation, and we should add anything relevant there. Leaving anything in the hacl-star repository would require maintenance.

Comment (franziskuskiefer) [reply doc/Obtaining.rst:91]: Yeah I agree. I just thought there might be interesting information here that you remove and that you could add to the two linked pages.

File: Makefile [hunk 3]
```diff
@@ -93,8 +93,8 @@ endif
 all: all-staged
 
 all-unstaged: compile-gcc-compatible compile-msvc-compatible \
-  compile-portable-gcc-compatible \
-  dist/wasm/package.json obj/libhaclml.cmxa
+  compile-portable-gcc-compatible compile-wasm \
```
Comment (msprotz) [inline Makefile:96]: this is going to call gcc, isn't it? see rule for `compile-%`

why not as before and take a dependency on dist/wasm/Makefile.basic, to force generation, but not compile in C?

Comment (pnmadelaine) [reply Makefile]: Thanks for the note, I replaced it with `dist/wasm/Makefile.basic`

File: dist/gcc-compatible/configure (hunk 2 above)
Comment (pnmadelaine) [reply dist/gcc-compatible/configure:230]: I don't know, @msprotz what do you think?

Comment (msprotz) [reply dist/gcc-compatible/configure:230]: I don't have a strong opinion either way... my hunch is if I'm working on the bindings I'll almost certainly end up with hacl-star + hacl-packages side-by-side with my testing loop doing build hacl, update hacl in packages, and build packages... so I don't think it makes sense to maintain an old build system in the hacl-star repository 

let's just make sure we're all very good about catching breakages... if the situation changes, we can always reintroduce some amount of build in HACL*

**Review by msprotz — APPROVED** [review]
lgtm 👍

**Pull request merged**